onnxruntime/core/framework/allocatormgr.cc

```diff
@@ -54,4 +61,38 @@ AllocatorPtr CreateAllocator(const AllocatorCreationInfo& info) {
   return AllocatorPtr(std::move(device_allocator));
 }
 
+
+// Update allocator in the provider if already present; ignore if not.
+void AllocatorManager::ReplaceAllocator(AllocatorPtr allocator) {
+  const auto& info = allocator->Info();
+  auto ite = mem_info_set_.find(info);
+  if (ite != mem_info_set_.end()) {
+    const int key = MakeKey(info.id, info.mem_type);
+    allocators_[key] = allocator;
+  }
```

Comment:
>} [](start = 2, length = 1)

Should ReplaceAllocator return failure or false if already exists? Seems it's silently skipping the input allocator.

Reply:
this one is for CreateAndRegisterAllocator API only. It's a copy from IExecutionProvider, as state in the comments, it updates allocator in the provider if already present; ignore if not.

---
In reply to: [564295727](https://github.com/microsoft/onnxruntime/pull/6332#discussion_r564295727) [](ancestors = 564295727)